curriculum/challenges/english/14-responsive-web-design-22/learn-css-colors-by-building-a-set-of-colored-markers/61695ab9f6ffe951c16d03dd.md

```diff
@@ -35,10 +35,46 @@ You should have a closing `body` tag.
 assert(code.match(/<\/body\s*>/i));
 ```
 
+There should only be one `head` opening tag.
+
+```js
+assert(code.match(/<head\s*>/ig).length === 1);
+```
+
+There should only be one `head` closing tag.
+
+```js
+assert(code.match(/<\/head\s*>/ig).length === 1);
+```
+
+There should only be one `body` opening tag.
+
+```js
+assert(code.match(/<body\s*>/ig).length === 1);
+```
+
+There should only be one `body` closing tag.
+
+```js
+assert(code.match(/<\/body\s*>/ig).length === 1);
+```
+
+Your `head` and `body` elements should be siblings.
+
+```js
+assert(code.match(/<head\s*>\s*<\/head\s*>/i));
+```
+
 Your `head` and `body` elements should be siblings.
 
 ```js
-assert(document.querySelector('head')?.nextElementSibling?.localName === 'body');
+assert(code.match(/<body\s*>\s*<\/body\s*>/i));
+```
```

Comment:
We have two tests here with the same text - and I don't think either of them really test what the text is asking for. Seems like we could maybe use a single test - maybe something like this:

`<head\s*>[\s\S]*<\/head\s*>\s*<body\s*>[\s\S]*<\/body\s*>|<body\s*>[\s\S]*<\/body\s*>\s*<head\s*>[\s\S]*<\/head\s*>`

Yea, it's long.

Reply:
Agreed, the hints for those two tests could be better. I have updated them to reflect what they are really checking (that the `head` and `body` elements should be empty).

I'm not sure your suggested regexp would work though. I think we always want the user to put the `head` before the `body`.